### PR TITLE
feat(auth): Add provider fallback for atmos auth login

### LINKED ISSUE #1915: `--identity` flag with `atmos auth console` is not passed through
### Describe the Bug

I'm having trouble passing in the `--identity` flag with `atmos auth console`.

The identity is properly set up, but setting the flag still prompts me to select an identity manually.

Image

### Expected Behavior

I should be able to open the AWS console in the browser without having to manually select which identity if I pass in the flag. 

The other `atmos auth *` commands are working as expected when passing in `--identity`.

For example, when I executed `atmos auth shell --identity ${identity}`, it enters into a shell with that identity properly configured, without having me select an identity.

Image

### Steps to Reproduce

See images above.

`atmos auth console --identity {identity}` prompts me to manually choose an identity even after I have already provided one via the flag.

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by osterman**
## what

- When `atmos auth login` is run without a `--provider` flag and no identities are configured, it now falls back to provider authentication
- Single providers are automatically selected without user input
- Multiple providers trigger an interactive selector (or error in non-interactive mode)
- Users no longer need to know about the `--provider` flag for initial authentication
- **Fix #1915**: `atmos auth console --identity {identity}` now correctly uses the provided identity instead of ignoring it

## why

- Improves user experience for first-time login with `auto_provision_identities` enabled
- Makes the system seamless after `atmos auth logout` (no need to rerun with `--provider` flag)
- Addresses feedback that `atmos auth login` should work without requiring explicit provider specification when only one provider exists
- **#1915 Fix**: The `--identity` flag with space-separated value (e.g., `--identity myid`) was being ignored due to Cobra's `NoOptDefVal` quirk. The fix uses `GetIdentityFromFlags` which parses `os.Args` directly to work around this issue.

## references

- Closes #1915
- User feedback: When `auto_provision_identities: true` is configured, the first `atmos auth login` command fails with "no identities available". User expects seamless fallback to provider auth when identities don't exist yet.

## Summary by CodeRabbit

* **New Features**
  * Auth login now falls back to provider authentication when no identities exist: auto-selects a single provider, prompts interactive selection for multiple providers, or requires --provider (-p) in non-interactive/CI.

* **Bug Fixes**
  * Improved identity-flag resolution and interactive selection edge-case handling; clearer error paths for missing providers or non-interactive selection.

* **Documentation**
  * Added CLI docs and blog post explaining provider-fallback behavior and usage.

* **Tests**
  * Expanded coverage for fallback flows, prompts, formatting, and identity-resolution.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: website/blog/2025-12-28-auth-login-provider-fallback.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Before: Required knowledge of the --provider flag
+atmos auth login --provider my-sso  # Had to know this
+
+# After: Just works
+atmos auth login  # Automatically uses the configured provider
+```
+
+## How to Use It
+
+No configuration changes needed. If you have a provider configured with `auto_provision_identities: true`, `atmos auth login` now works out of the box:
```
Comment (oycyc) [inline website/blog/2025-12-28-auth-login-provider-fallback.mdx:36]: Awesome! ❤️

**Review by aknysh — APPROVED** [review]
